tensorflow_addons/rnn/cell.py

```diff
@@ -799,3 +799,50 @@ def get_config(self):
         }
         base_config = super().get_config()
         return {**base_config, **config}
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class ZoneoutLSTMCell(keras.layers.LSTMCell):
+    """LSTM cell with recurrent zoneout.
+
+    https://arxiv.org/abs/1606.01305
+    """
+
+    @typechecked
+    def __init__(
+        self,
+        units: int,
+        zoneout_h: float = 0,
+        zoneout_c: float = 0,
+        seed: int = None,
+        **kwargs
+    ):
+        """
+        """
+        super().__init__(units, **kwargs)
+        self.zoneout_h = zoneout_h
+        self.zoneout_c = zoneout_c
+        self.seed = seed
+
+    def _zoneout(self, t, tm1, rate, training):
+        dt = tf.cast(
+            tf.random.uniform(t.shape, seed=self.seed) >= rate * training, t.dtype
+        )
+        return dt * t + (1 - dt) * tm1
+
+    def call(self, inputs, states, training=None):
```

Comment:
In order to get the `training` value when it's None, try first with  `base_layer_utils.call_context().training`,  which is a more specific approach. Then, if not found, use keras backend as you were doing. 
You can use [this](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/backend.py#L4364) as a reference.

Reply:
How does your approach outperform the current one? It's just a fallback. Your fallback still requires that fallback. :) I'm even not sure that we need it here but just peeked it from Tensorflow's Dropout implementation.

Reply:
the `call_context` function seemed to be a more specific function. The approach I suggested is the one used by default, it is implemented here: [in_train_phase](https://github.com/tensorflow/tensorflow/blob/a3aec8d37ec6dcd7e82a68067162f492be00944f/tensorflow/python/keras/backend.py#L4346).
However, I see your reference from Dropout [implementation](https://github.com/tensorflow/tensorflow/blob/2b96f3662bd776e277f86997659e61046b56c315/tensorflow/python/keras/layers/core.py#L198). Seems ok for me, thanks again :).